code/modules/reagents/Chemistry-Holder.dm

```diff
@@ -775,6 +775,16 @@
 	chem_temp = round(chem_temp)
 	handle_reactions()
 
+/datum/reagent/proc/purge(mob/living/carbon/M)
```

Comment:
Make this `/datum/reagents/proc/purge(purge_rate)` (reagent**s**, plural and not singular, be mindful), since it should be a holder proc. Take note to to remove all `M.reagents` within and add `reagents.` prefixes on the calls.